src/PostgREST/App.hs

```diff
@@ -193,28 +192,23 @@ app dbStructure proc cols conf apiRequest =
                     (iPreferRepresentation apiRequest) []
               row <- H.statement (toS $ pjRaw pJson) stm
               let (_, queryTotal, _, body) = row
-
-                  updateIsNoOp       = S.null cols
-                  contentRangeHeader = contentRangeH 0 (queryTotal - 1) $
-                                          if shouldCount then Just queryTotal else Nothing
-                  minimalHeaders     = [contentRangeHeader]
-                  fullHeaders        = toHeader contentType : minimalHeaders
-
+                  updateIsNoOp = S.null cols
+                  r = contentRangeH 0 (queryTotal - 1) $
+                        if shouldCount then Just queryTotal else Nothing
+                  headers | iPreferRepresentation apiRequest == Full = [toHeader contentType, r]
+                          | otherwise                                = [r]
                   status | queryTotal == 0 && not updateIsNoOp      = status404
                          | iPreferRepresentation apiRequest == Full = status200
                          | otherwise                                = status204
-
-              case (contentType, iPreferRepresentation apiRequest) of
-                (CTSingularJSON, Full)
-                      | queryTotal == 1 -> return $ responseLBS status fullHeaders (toS body)
-                      | otherwise       -> HT.condemn >> (return . errorResponseFor . singularityError) queryTotal
-
-                (_, Full) ->
-                  return $ responseLBS status fullHeaders (toS body)
-
-                (_, _) ->
-                  return $ responseLBS status minimalHeaders mempty
-
```

Comment:
This previous case expression was much easier to read for me. Could you keep the same structure and apply the new logic? I think it'd be a matter of wildcard matching the `iPreferRepresentation`:

```hs
case (contentType, iPreferRepresentation apiRequest) of
  (CTSingularJSON, _) -> .... 
```

Reply:
This variant of code is more corresponding with other parts.

Reply:
@dwagin The other parts could be refactored later. Variables like `r` are not really meaningful and it's important that we keep improving readability.

Reply:
Naming changed.

Reply:
IMHO switch/case in this case less readable and requires code duplication.

Reply:
It reads better now. Thanks!